### ISSUE TITLE
Intense lag dragging around XWayland windows.

### ISSUE THREAD

**Issue by Surferlul** [alecdotninja/no-titlebar-when-maximized#5]
When dragging around X11 applications in Wayland with this extension enabled it is almost unusable

**Comment by alecdotninja** [comment]
I do not have this issue on Arch with GNOME 42 under Wayland. Does disabling the extension resolve the problem?

**Comment by Surferlul** [comment]
Yes,it does. I was using Fedora 36 (same as the duplicate). It's not a hardware problem, because it happened on both my workstation (VM with 14 5950x cores, 6900xt PCIe passthrough, 28GB RAM), and my laptop ([Lenovo Yoga slim 7 pro](https://www.lenovo.com/gb/en/laptops/yoga/yoga-slim-series/Yoga-Slim-7-Pro-Gen-6-16-inch-AMD/p/LEN101Y0005), 5800H, integrated graphics & Nvidia 3050, 16GB RAM), unless it is AMD specific problem. It was worse on my laptop, but still really bad on my workstation.

**Comment by alecdotninja** [comment]
@Surferlul I've added some debouncing and more logging to the extension in the latest version. If you are still seeing the issue, I am curious if anything interesting is logged.

**Comment by alecdotninja** [comment]
Unfortunately, the changes I made to try to address this caused other problems, so I have reverted them.

If someone who is experiencing the issue has time to debug, a PR is more than welcome! :smile:
